### ISSUE TITLE
plot(validator) produces "Error in as.double(y): cannot coerce type 'S4' to vector of type 'double'"

### ISSUE THREAD

**Issue by Plebejer** [data-cleaning/validate#91]
Hi Mark,

plotting a validator object does also not work in my work environment. The function call is simply `plot(validator)`. The error message is 

> Error in as.double(y) : 
> cannot coerce type 'S4' to vector of type 'double'

 Here is the session info:

    R version 3.5.0 (2018-04-23)
    Platform: x86_64-w64-mingw32/x64 (64-bit)
    Running under: Windows 7 x64 (build 7601) Service Pack 1
    
    Matrix products: default
    
    locale:
    [1] LC_COLLATE=German_Germany.1252  LC_CTYPE=German_Germany.1252    LC_MONETARY=German_Germany.1252
    [4] LC_NUMERIC=C                    LC_TIME=German_Germany.1252    
    
    attached base packages:
    [1] stats     graphics  grDevices utils     datasets  methods   base     
    
    other attached packages:
    [1] bindrcpp_0.2.2    readtext_0.71     lubridate_1.7.4   rlang_0.2.1       fs_1.2.3          validate_0.2.5   
    [7] forcats_0.3.0     stringr_1.3.1     dplyr_0.7.6       readr_1.1.1       tidyr_0.8.1       tibble_1.4.2     
    [13] ggplot2_3.0.0     tidyverse_1.2.1   readxl_1.1.0      magrittr_1.5      purrr_0.2.5       data.table_1.11.4
    
    loaded via a namespace (and not attached):
    [1] Rcpp_0.12.18     cellranger_1.1.0 pillar_1.3.0     compiler_3.5.0   plyr_1.8.4       bindr_0.1.1     
    [7] sys_1.5          tools_3.5.0      antiword_1.1     bit_1.1-14       jsonlite_1.5     nlme_3.1-137    
    [13] gtable_0.2.0     lattice_0.20-35  pkgconfig_2.0.1  cli_1.0.0        rstudioapi_0.7   yaml_2.1.19     
    [19] haven_1.1.2      withr_2.1.2      xml2_1.2.0       httr_1.3.1       hms_0.4.2        bit64_0.9-7     
    [25] grid_3.5.0       tidyselect_0.2.4 glue_1.3.0       R6_2.2.2         modelr_0.1.2     settings_0.2.4  
    [31] backports_1.1.2  scales_0.5.0     rvest_0.3.2      assertthat_0.2.0 colorspace_1.3-2 stringi_1.1.7   
    [37] lazyeval_0.2.1   munsell_0.5.0    broom_0.5.0      crayon_1.3.4    

The same error occurs on my personal Linux machine.

**Comment by Plebejer** [comment]
And here is the `sessionInfo()` from my personal Linux machine:

> R version 3.5.1 (2018-07-02)
> Platform: x86_64-pc-linux-gnu (64-bit)
> Running under: Ubuntu 18.04.1 LTS
> 
> Matrix products: default
> BLAS: /opt/microsoft/ropen/3.5.1/lib64/R/lib/libRblas.so
> LAPACK: /opt/microsoft/ropen/3.5.1/lib64/R/lib/libRlapack.so
> 
> locale:
>  [1] LC_CTYPE=de_DE.UTF-8       LC_NUMERIC=C               LC_TIME=de_DE.UTF-8       
>  [4] LC_COLLATE=de_DE.UTF-8     LC_MONETARY=de_DE.UTF-8    LC_MESSAGES=de_DE.UTF-8   
>  [7] LC_PAPER=de_DE.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
> [10] LC_TELEPHONE=C             LC_MEASUREMENT=de_DE.UTF-8 LC_IDENTIFICATION=C       
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] validatetools_0.4.3  errorlocate_0.1.3    validate_0.2.5       data.table_1.11.4   
> [5] shiny_1.1.0          RevoUtils_11.0.1     RevoUtilsMath_11.0.0
> 
> loaded via a namespace (and not attached):
>  [1] Rcpp_0.12.18          crayon_1.3.4          digest_0.6.15         later_0.7.3          
>  [5] mime_0.5              R6_2.2.2              jsonlite_1.5          xtable_1.8-2         
>  [9] magrittr_1.5          rlang_0.2.1           settings_0.2.4        rstudioapi_0.7       
> [13] promises_1.0.1        lpSolveAPI_5.5.2.0-17 tools_3.5.1           httpuv_1.4.5         
> [17] yaml_2.2.0            compiler_3.5.1        htmltools_0.3.6 

Is there anything else I could provide to help understanding this behaviour?

**Comment by edwindj** [comment]
Hi Mathias,

Is this your complete code? Or is there a certain rule set you are trying to plot?
if I do a 

```r
rules <- validator(x > 0)
plot(rules)
```
if works fine.

If I do 
```r
plot(validator) # which is a strange thing to do, plotting the function...
```
I get
```
Error in x@expr[[1]] : object of type 'symbol' is not subsettable
```
But this is fine, because that is standard R behavior. (e.g. `plot(lm)` is also not working...)

**Comment by Plebejer** [comment]
Hi Edwin,

sorry for not providing a MWE initially. With `plot(validator)` I wanted to express that I feed a validator object into `plot()`.

Here is a MWE which reliably reproduces the error from a new R session. I actually receive the error for both `plot()` calls.

    library(validate)
    data <- data.frame(A = 1)
    rule <- validator(A > 0)
    cf <- confront(data, rule)
    
    plot(rule)
    plot(cf)

    Error in as.double(y) : 
      cannot coerce type 'S4' to vector of type 'double'

I think the error is not related to the data or the rule set since this MWE is totally unrelated to the other settings in which I receive the error.

**Comment by edwindj** [comment]
Weird, your code happens to work on my machine (but that is a Linux machine, will test asap on Windows)

**Comment by KostasSoul** [comment]
Hi,

I had the same issue when I tried to plot a "validator" object.
I don't know if that helps, but I re-installed the latest version of the package (0.2.6) and eventually worked just fine. I also use Windows.

Cheers
Kostas

**Comment by edwindj** [comment]
@KostasSoul: Thanks for your report!

Since reinstalling seems to solve the problem I suggest to close the issue. Do not hesitate to reopen it if the issue returns!

**Comment by Plebejer** [comment]
Yes, can be closed. I figured out that I did not use the most recent version 0.2.6, but 0.2.5. A simple reinstall did not help in my case since I'm using Microsoft R Open and I had to set a more recent snapshot date in order to install version 0.2.6. 

In case anyone has the same problem in the future: Another snapshot date can be set via 

`options(repos = c(CRAN = "https://mran.revolutionanalytics.com/snapshot/YYYY-MM-DD"))`
